clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -1358,6 +1359,8 @@ class ConstExprEmitter :
   }
 
   llvm::Constant *VisitStringLiteral(StringLiteral *E, QualType T) {
+    if (!isa<ConstantArrayType>(T.getDesugaredType(CGM.getContext())))
```

Comment:
Alternatively this could be:

`if (isa<LValueReferenceType>)`

any preference/thoughts @efriedma-quic ?